rosidl_parser/rosidl_parser/definition.py

```diff
@@ -107,170 +167,231 @@ def __eq__(self, other):
         return super().__eq__(other) and self.name == other.name
 
 
-class NamespacedType(BaseType):
+class NamespacedType(AbstractNestableType):
     """A type identified by a name in a namespaced scope."""
 
     __slots__ = ('namespaces', 'name')
 
-    def __init__(self, namespaces, name):
+    def __init__(self, namespaces: Iterable[str], name: str):
         """
         Constructor.
 
-        :param list[str] namespaces: the names of nested namespaces identifying
-          a specific scope
-        :param str name: the name of the type within that scope
+        :param namespaces: the names of nested namespaces identifying a
+          specific scope
+        :param name: the name of the type within that scope
         """
         super().__init__()
         self.namespaces = namespaces
         self.name = name
 
+    def namespaced_name(self) -> Tuple[str, ...]:
+        return (*self.namespaces, self.name)
+
     def __eq__(self, other):
         return super().__eq__(other) and \
             self.namespaces == other.namespaces and self.name == other.name
 
 
-class BaseString(BaseType):
-    """The base class of string types."""
+class AbstractGenericString(AbstractNestableType):
+    """The abstract base class of all string types."""
+
+    __slots__ = ()
+
+    def has_maximum_size(self):
+        raise NotImplementedError('Only implemented in subclasses')
+
+
+class AbstractString(AbstractGenericString):
```

Comment:
An in-between abstract class for 8-bit strings (bounded and unbounded). There a similar one for 16-bit strings below. It just ease the logic instead of having to check for two concrete types each time.